packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -72,7 +55,7 @@ function NavigationMenuSelector( {
 	const menuChoices = useMemo( () => {
 		return (
 			navigationMenus?.map( ( { id, title, status }, index ) => {
-				const label = buildMenuLabel(
+				const label = buildNavigationLabel(
```

Comment:
The destructuring of the navigationMenu object in here means we need to make more changes to the code to pass the whole object. I'm not sure it's worth it. By only passing the properties we need we keep the function clearer and less susceptible to side effects.